test/NuGet.Core.Tests/Microsoft.Build.NuGetSdkResolver.Test/GlobalJsonReader_Tests.cs

```diff
@@ -11,23 +11,123 @@
 
 namespace Microsoft.Build.NuGetSdkResolver.Test
 {
+    /// <summary>
+    /// Unit tests for the <see cref="GlobalJsonReader" /> class.
+    /// </summary>
     public class GlobalJsonReaderTests
     {
+        /// <summary>
+        /// Verifies that <see cref="GlobalJsonReader.GetMSBuildSdkVersions(Framework.SdkResolverContext)" /> ignores duplicates in the msbuild-sdks section and uses the last specified version.
+        /// </summary>
         [Fact]
-        public void EmptyGlobalJson()
+        public void GetMSBuildSdkVersions_IgnoresDuplicates_WhenGlobalJsonContainsDuplicates()
+        {
+            var expectedVersions = new Dictionary<string, string>
+            {
+                {"Sdk1", "3.0.0"},
+                {"Sdk2", "2.0.0"},
+            };
+
+            using (var testDirectory = TestDirectory.Create())
+            {
+                File.WriteAllText(
+                    Path.Combine(testDirectory, GlobalJsonReader.GlobalJsonFileName),
+                    @"{
+  // This is a comment
+  ""msbuild-sdks"": {
+    ""Sdk1"": ""1.0.0"",
+    ""Sdk2"": ""2.0.0"",
+    ""Sdk1"": ""3.0.0"",
+  }
+}
+");
+
+                var context = new MockSdkResolverContext(testDirectory);
+
+                GlobalJsonReader.GetMSBuildSdkVersions(context).Should().Equal(expectedVersions);
+            }
+        }
+
+        /// <summary>
+        /// Verifies that <see cref="GlobalJsonReader.GetMSBuildSdkVersions(Framework.SdkResolverContext)" /> logs a message when the specified global.json contains invalid JSON.
+        /// </summary>
+        [Fact]
+        public void GetMSBuildSdkVersions_LogsMessage_WhenGlobalJsonContainsInvalidJson()
+        {
+            var expectedVersions = new Dictionary<string, string>
```

Comment:
Is this var referenced anywhere?

Reply:
It is currently passed to `WriteGlobalJson()` so that those versions end up in the file, but then some invalid JSON is also included in there.  So we expect the parser to log a message and return `null`.